aiidalab/app.py

```diff
@@ -652,10 +656,20 @@ def _default_busy(self):  # pylint: disable=no-self-use
     def _show_busy(self):
         """Apply this decorator to indicate that the app is busy during execution."""
         self.set_trait("busy", True)
+
+        # we need to use a lock here, because the busy trait is not thread-safe
+        # we may use _show_busy in different threads, e.g. when installing and auto-status refresh
+        with self._busy_count_lock:
+            self._busy_count += 1
```

Comment:
```suggestion
        # we need to use a lock here, because the busy trait is not thread-safe
        # we may use _show_busy in different threads, e.g. when installing and auto-status refresh
        with self._busy_count_lock:
            self._busy_count += 1
            self.set_trait("busy", True)
```

Reply:
Thanks. If it is within the lock I think the order does not matter. I will test this PR now.